driftbase/models/db.py

```diff
@@ -114,7 +112,35 @@ def check_password(self, password):
 
 tbl_user_identity = UserIdentity.__table__
 
-
+class PlayerRichPresence:
+    """
+    Rich presence information for a particular user.
+    @see PlayerRichPresenceSchema
+    """
+    
+    game_mode = ""
+    map_name = ""
+    is_online = False
+    is_in_game = False
+
+    def __init__(self, player_id: int, is_online: bool):
+        self.is_online = is_online
+
+        try:
+            match_player : MatchPlayer = g.db.query(MatchPlayer) \
+                .filter(MatchPlayer.player_id == player_id) \
+                .one()
+            
+            match : Match = g.db.query(Match) \
+                .filter(Match.match_id == match_player.match_id) \
+                .one()
+
+            self.is_in_game = match_player.status == "active" # Always true?
```

Comment:
Status is update a bit more frequently when using the legacy match maker I believe, but could be "banned" in the future.

Reply:
The actual valid states that are used may be unknown, since it can be set by the battle servers too, I believe

Reply:
Yeah we may need to keep an eye on this "status" field, and whether it's actually a reliable source of information for the matches status. I think for now at least, 'active' is reliably used when the match is ongoing.